src/client/pmi1.c

```diff
@@ -59,20 +61,37 @@ static int pmi_init = 0;
 
 int PMI_Init(int *spawned)
 {
+    pmix_status_t rc = PMIX_SUCCESS;
     pmix_value_t *val;
-    pmix_status_t rc;
+    pmix_proc_t proc;
+    pmix_info_t info[2];
+    bool  val_optinal = 1;
+    int val_timeout = PMIX_TIMEOUT_VALUE;
 
     if (PMIX_SUCCESS != PMIx_Init(&myproc)) {
         return PMI_ERR_INIT;
     }
 
+    /* getting internal key requires special rank value */
+    memcpy(&proc, &myproc, sizeof(myproc));
+    proc.rank = PMIX_RANK_UNDEF;
+
+    /* set controlling parameters
+     * PMIX_OPTIONAL - expect that these keys should be available on startup
+     * PMIX_TIMEOUT - meet legacy PMI requirement
+     */
+    PMIX_INFO_CONSTRUCT(&info[0]);
+    PMIX_INFO_CONSTRUCT(&info[1]);
+    PMIX_INFO_LOAD(&info[0], PMIX_OPTIONAL, &val_optinal, PMIX_BOOL);
+    PMIX_INFO_LOAD(&info[1], PMIX_TIMEOUT, &val_timeout, PMIX_INT);
```

Comment:
@rhc54 A absolutelly agree that nobody can predict a time get operation is processed. 
My motivation to use timeout in PMI/PMI2 client is:
- Now calling get operation requesting non existing key for specific rank brings stuck of application (example PMI2_Info_GetNodeAttr()).
- pmi2/pmi do not have a possibility to use PMIx_Get_nb.
- We do not know exactly what keys are mandatory for PMI/PMI2
- I prefer to return NOT_FOUND status instead of blocking situation for legacy PMI/PMI2 api.
As as result 
I am considering timeout usage as a way of replacing /PMIx_Get_nb/ - /user defined wait call/ inside PMI/PMI2 clients.

What timeout value do you suggest to set?